src/components/molecules/Visualizer/Engine/Cesium/core/Indicator.tsx

```diff
@@ -0,0 +1,133 @@
+import useTransition, { TransitionStatus } from "@rot1024/use-transition";
+import { BoundingSphere, Cartesian3, SceneTransforms, Cartesian2 } from "cesium";
+import React, { useEffect, useState } from "react";
+import { useCesium } from "resium";
+
+import Icon from "@reearth/components/atoms/Icon";
+import { styled } from "@reearth/theme";
+
+import { SceneProperty } from "../../ref";
+import { useIcon } from "../common";
+
+export type Props = {
+  className?: string;
+  property: SceneProperty | undefined;
+};
+
+export default function Indicator({ className, property }: Props): JSX.Element | null {
+  const { viewer } = useCesium();
+  const [isVisible, setIsVisible] = useState(true);
+  const [pos, setPos] = useState<Cartesian2>();
+
+  const transiton = useTransition(!!pos && isVisible, 500, {
+    mountOnEnter: true,
+    unmountOnExit: true,
+  });
+  const { indicator_type, indicator_image, indicator_img_scale } = property?.indicator ?? {};
+  const [img, w, h] = useIcon({ image: indicator_image, imageSize: indicator_img_scale });
+
+  useEffect(() => {
+    if (!viewer) return;
+    const handleTick = () => {
+      if (viewer.isDestroyed()) return;
+      const selected = viewer.selectedEntity;
+      if (
+        !selected ||
+        !selected.isShowing ||
+        !selected.isAvailable(viewer.clock.currentTime) ||
+        !selected.position
+      ) {
+        setIsVisible(false);
+        return;
+      }
+
+      // https://github.com/CesiumGS/cesium/blob/1.94/Source/Widgets/Viewer/Viewer.js#L1839
+      let position: Cartesian3 | undefined = undefined;
+      const boundingSphere = new BoundingSphere();
+      const state = (viewer.dataSourceDisplay as any).getBoundingSphere(
+        selected,
+        true,
+        boundingSphere,
+      );
+      // https://github.com/CesiumGS/cesium/blob/main/Source/DataSources/BoundingSphereState.js#L24
+      if (state !== 2 /* BoundingSphereState.FAILED */) {
+        position = boundingSphere.center;
+      } else if (selected.position) {
+        position = selected.position.getValue(viewer.clock.currentTime, position);
+      }
+
+      if (position) {
+        const pos = SceneTransforms.wgs84ToWindowCoordinates(viewer.scene, position);
+        setPos(pos);
+        setIsVisible(true);
+      } else {
+        setIsVisible(false);
+      }
+    };
+
+    viewer.clock.onTick.addEventListener(handleTick);
+    return () => {
+      if (viewer.isDestroyed()) return;
+      viewer.clock.onTick.removeEventListener(handleTick);
+    };
+  }, [viewer]);
+  // console.log( indicator_type );
+  return transiton !== "unmounted" && pos ? (
+    indicator_type === "crosshair" ? (
+      <StyledIcon
+        icon="crosshair"
+        className={className}
+        transition={transiton}
+        style={{ left: pos.x + "px", top: pos.y + "px" }}
+      />
+    ) : indicator_type === "default" ? (
```

Comment:
Shouldn't the cesium default be the fallback option, not the Image?

Reply:
But here the logic goes:
- is it crosshair? show crosshair. if not,
- is it default? show default. if not,
- show Image component

So the fallback is the Image component (`indicator_type === "custom"` right?). Can you make it so that `indicator_type === "custom"` and Image component is checked, and then it fallsback to the default?